### PR TITLE
Upload gzipped contents

### PR DESCRIPTION
**Pull request by jcrugzz**
## Summary
Fixes the problem we had where we werent uploading the gzipped content that we were given by workers-factory in carpenterd-worker

## Changelog
* make the change to upload compressed contents

## Test Plan
tests work with addition

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -405,7 +411,7 @@ BFFS.prototype.publish = function publish(spec, options, fn) {
             extension: file.extension,
             filename: file.filename,
             fingerprint: print,
-            url: file.url
+            url: gz ? file.url : file.compressedUrl
```
Comment (fritzmonkey) [inline index.js]: This seems reversed to me. If `gz` is `false` we want the `compressedUrl`?

File: index.js [hunk 2]
```diff
@@ -346,6 +346,12 @@ BFFS.prototype.publish = function publish(spec, options, fn) {
             fn(err);
           });
         },
+        file.compressed && function (fn) {
+          cdn.upload(file.compressed, bff.key(file, 'file') + '.gz', (err, url) => {
+            file.compressedUrl = url;
```
Comment (fritzmonkey) [inline index.js]: Do we want to set this regardless of if there is an error?

**Comment by jcrugzz** [comment]
@decompil3d for context https://github.com/warehouseai/cdnup/pull/13 is what sets the things when we upload to s3. We can verify on this end though

File: index.js (hunk 1 above)
Comment (jcrugzz) [reply index.js]: @fritzmonkey thanks, you are correct

File: index.js (hunk 2 above)
Comment (jcrugzz) [reply index.js]: @fritzmonkey my view was that it didnt matter much since error short circuits

**Pull request merged**